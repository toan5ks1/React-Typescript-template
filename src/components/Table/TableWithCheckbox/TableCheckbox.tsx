import { StandardProps } from "src/typings/react-component";

import { Checkbox, CheckboxProps } from "@mui/material";

interface TableCheckboxProps extends StandardProps {
    checkboxProps: CheckboxProps;
}

const TableCheckbox = (props: TableCheckboxProps) => {
    const { checkboxProps, ...rest } = props;

    return (
        //TODO: Need implement CheckboxBase in the future
        <Checkbox
            {...rest}
            style={{
                padding: 0,
                width: 18,
                height: 18,
            }}
            color="primary"
            {...checkboxProps}
        />
    );
};

export default TableCheckbox;
