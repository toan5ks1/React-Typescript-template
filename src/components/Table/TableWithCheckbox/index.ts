export { default } from "./TableWithCheckbox";
export * from "../table-types";
