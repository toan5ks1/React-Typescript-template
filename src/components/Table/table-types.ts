import { ComponentProps, CSSProperties, ElementType, HTMLAttributes, Key, ReactNode } from "react";

import { StandardProps } from "src/typings/react-component";

import { Table, TableCellProps } from "@mui/material";

import { PaginationWithTotal } from "src/hooks/data/useQueryWithPagination";

export interface CellProps extends TableCellProps {
    "data-testid"?: string;
    style?: CSSProperties;
    headerStyle?: CSSProperties;
}

export interface TableColumn<T extends object = any> {
    key: string;
    title?: ReactNode;
    emptyValue?: string;
    testId?: string;
    cellProps?: CellProps;
    dataIndex?: string;
    render?: (record: T, dataIndex?: number) => ReactNode;
}

export type TableIndex = boolean | Pick<CSSProperties, "width">;

export enum TableBorderStyle {
    all = "all",
    horizontal = "horizontal",
    none = "none",
}

export interface RowKeyFunc {
    (record: object): string;
}

export type StyleKeys = {
    root: CSSProperties;
    container: CSSProperties;
    table: CSSProperties;
    row: CSSProperties;
    header: CSSProperties;
    footer: CSSProperties;
};

export interface TableCheckboxSelectType {
    onSelectAllClick: (isSelectAll: boolean) => void;
    hasData: boolean;
    hasChecked: boolean;
    isCheckedAll: boolean;
}

export interface TableBaseHeaderProps extends StandardProps {
    visible: boolean;
    border: CSSProperties;
    columns: TableColumn[];
    withIndex?: TableIndex;
    render?: (columns: TableColumn[]) => ReactNode;
}

export interface TableBaseBodyProps<TData>
    extends Pick<
        TableBaseRowProps<TData>,
        "border" | "rowKey" | "columns" | "hover" | "withIndex"
    > {
    data: TData[];
    loading: boolean;
    noDataMessage?: string | ReactNode;
    isSelected?: Function;
    customBody?: ReactNode;
    skeCount?: number;
    skeHeight?: number;
    pagination?: PaginationWithTotal;
    styleIndexCol?: TableBaseRowProps<TData>["styleIndexCol"];
}

export interface TableBaseRowProps<TData>
    extends Omit<HTMLAttributes<HTMLTableRowElement>, "onSelect"> {
    dataIndex: number;
    record: TData;
    columns: TableColumn[];
    selected?: boolean;
    rowKey: string | RowKeyFunc;
    hover?: boolean;
    border: CSSProperties;
    withIndex?: TableIndex;
    pagination?: PaginationWithTotal;
    render?: (columns: TableColumn[]) => ReactNode;
    styleIndexCol?: CSSProperties;
}

export interface TableBaseCellProps extends CellProps {
    border?: Pick<
        CSSProperties,
        "border" | "borderLeft" | "borderTop" | "borderRight" | "borderBottom"
    >;
    maxColumnIndex?: number;
    columnIndex?: number;
}

export interface TableBaseFooterProps {
    pagination: PaginationWithTotal;
    labelRowsPerPage?: string;
    rowsPerPageOptions?: number[];
    style?: CSSProperties;
}

export interface TableBaseProps<TData> extends StandardProps {
    columns: TableColumn[];
    data: TData[];
    showHeader: boolean;
    border: keyof typeof TableBorderStyle; // Support 3 border style
    body: Omit<TableBaseBodyProps<TData>, "data" | "columns" | "border">;
    footer?: TableBaseFooterProps;
    styles?: Partial<StyleKeys>; // Support custom style by using style attribute
    tableProps?: object | ComponentProps<typeof Table>; // Sticky header, put data-id...
    component: ElementType;
    isSelected?: (key: Key) => boolean;
    errorMessage?: string | ReactNode;
    withIndex?: TableIndex;
    styleIndexCol?: TableBaseRowProps<TData>["styleIndexCol"];
}

export interface TableHeaderWithCheckboxProps extends Omit<TableBaseHeaderProps, "render"> {
    select?: TableCheckboxSelectType;
}

export interface TableBodyWithCheckboxProps<TData>
    extends TableBaseBodyProps<TData>,
        Pick<TableRowWithCheckboxProps<TData>, "onSelect" | "handleClickCheckbox">,
        Pick<TableWithCheckboxProps<TData>, "hasDraftProperty"> {}

export interface TableRowWithCheckboxProps<TData>
    extends Omit<TableBaseRowProps<TData>, "render">,
        Pick<TableWithCheckboxProps<TData>, "hasDraftProperty"> {
    handleClickCheckbox?: (record: TData) => void;
    onSelect?: (record: TData[]) => void;
}

export interface TableWithCheckboxProps<TData> extends Omit<TableBaseProps<TData>, "isSelected"> {
    listSelectedItems?: TData[]; // Allow pass a list with selected item from outside
    onSelect?: (record: TData[]) => void; // Support checkbox each row
    hasDraftProperty?: boolean;
}

export interface TableBodyWithRowSpanProps<TData> extends Omit<TableBaseBodyProps<TData>, "data"> {
    data: { [group: string]: TData[] };
}

export interface TableWithRowSpanProps<TData>
    extends Omit<TableBaseProps<TData>, "columns" | "data"> {
    headerColumns: TableColumn[];
    dataColumns: TableColumn[];
    data: { [group: string]: TData[] };
}
