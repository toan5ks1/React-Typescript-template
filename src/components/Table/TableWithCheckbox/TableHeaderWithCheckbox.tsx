import { Box } from "@mui/material";
import TypographyBase from "src/components/Typographys/TypographyBase";

import TableBaseCell from "../TableBaseCell";
import TableBaseHeader from "../TableBaseHeader";
import { TableColumn, TableHeaderWithCheckboxProps } from "../table-types";
import TableCheckbox from "./TableCheckbox";

import useTranslate from "src/hooks/useTranslate";

const TableHeaderWithCheckbox = (props: TableHeaderWithCheckboxProps) => {
    const { select, ...tableBaseHeaderProps } = props;
    const { border } = tableBaseHeaderProps;
    const t = useTranslate();
    const chooseAllItemCheckBox = t("ra.common.action.selectAllItems");

    const renderCell = (column: TableColumn, index: number) => {
        if (select && index === 0) {
            return (
                <Box display="flex" alignItems="center">
                    <Box mr={1.5} display="flex">
                        <TableCheckbox
                            data-testid="TableHeaderWithCheckbox__checkboxHeader"
                            checkboxProps={{
                                indeterminate: select.hasChecked && !select.isCheckedAll,
                                checked: select.isCheckedAll && select.hasData,
                                // onChange not working when click on the border
                                onClick: () => select.onSelectAllClick(!select.isCheckedAll),
                                disabled: !select.hasData,
                                inputProps: { "aria-label": chooseAllItemCheckBox },
                            }}
                        />
                    </Box>
                    <TypographyBase
                        data-testid="TableHeaderWithCheckbox__cellHeader"
                        variant="subtitle2"
                    >
                        {column.title}
                    </TypographyBase>
                </Box>
            );
        }

        return (
            <TypographyBase data-testid="TableHeaderWithCheckbox__cellHeader" variant="subtitle2">
                {column.title}
            </TypographyBase>
        );
    };

    return (
        <TableBaseHeader
            render={(columns) =>
                columns.map((column, index) => {
                    const { cellProps } = column;
                    const columnCellProps = Object.assign({}, cellProps);
                    delete columnCellProps.headerStyle;

                    return (
                        <TableBaseCell
                            {...columnCellProps}
                            style={
                                cellProps?.headerStyle ? cellProps?.headerStyle : cellProps?.style
                            }
                            key={column.key}
                            border={border}
                            maxColumnIndex={columns.length - 1}
                            columnIndex={index}
                        >
                            {renderCell(column, index)}
                        </TableBaseCell>
                    );
                })
            }
            {...tableBaseHeaderProps}
        />
    );
};

export default TableHeaderWithCheckbox;
