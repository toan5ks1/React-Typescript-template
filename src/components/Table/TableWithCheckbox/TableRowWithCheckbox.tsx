import get from "lodash/get";

import { Box } from "@mui/material";

import TypographyBase from "../../Typographys/TypographyBase";
import TableBaseCell from "../TableBaseCell";
import TableBaseRow from "../TableBaseRow";
import { TableColumn, TableRowWithCheckboxProps } from "../table-types";
import TableCheckbox from "./TableCheckbox";

const TableRowWithCheckbox = <T extends object>(props: TableRowWithCheckboxProps<T>) => {
    const { onSelect, handleClickCheckbox, ...tableBaseRowProps } = props;
    const { dataIndex, border, record, selected, hasDraftProperty } = tableBaseRowProps;

    const renderCell = (column: TableColumn, index: number) => {
        // If table has checkbox list. We will show checkbox in first column
        if (onSelect && index === 0) {
            return (
                <Box display="flex" alignItems="center">
                    <Box mr={1.5} display="flex">
                        <TableCheckbox
                            data-testid="TableRowWithCheckbox__checkboxRow"
                            checkboxProps={{
                                checked: selected,
                                onClick: () => {
                                    handleClickCheckbox?.(record);
                                },
                                disabled: !record?.hasOwnProperty("isDraft") && hasDraftProperty,
                            }}
                        />
                    </Box>
                    {typeof column.render === "function" ? (
                        column.render(record, dataIndex)
                    ) : (
                        <TypographyBase variant="body2" data-testid={column.testId}>
                            {column.dataIndex ? get(record, column.dataIndex) : column.emptyValue}
                        </TypographyBase>
                    )}
                </Box>
            );
        }

        if (typeof column.render === "function") {
            return column.render(record, dataIndex);
        }

        return (
            <TypographyBase variant="body2" data-testid={column.testId}>
                {column.dataIndex ? get(record, column.dataIndex) : column.emptyValue}
            </TypographyBase>
        );
    };

    return (
        <TableBaseRow
            render={(columns) =>
                columns.map((column, index) => {
                    const { cellProps } = column;
                    const columnCellProps = Object.assign({}, cellProps);
                    delete columnCellProps.headerStyle;

                    return (
                        <TableBaseCell
                            key={column.key}
                            {...columnCellProps}
                            border={border}
                            style={column.cellProps?.style}
                            maxColumnIndex={columns.length - 1}
                            columnIndex={index}
                        >
                            {renderCell(column, index)}
                        </TableBaseCell>
                    );
                })
            }
            {...tableBaseRowProps}
        />
    );
};

export default TableRowWithCheckbox;
