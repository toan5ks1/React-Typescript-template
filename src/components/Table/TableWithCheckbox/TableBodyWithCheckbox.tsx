import TableBaseBody from "../TableBaseBody";
import { TableBodyWithCheckboxProps } from "../table-types";
import { getRowKey } from "../utils";
import TableRowWithCheckbox from "./TableRowWithCheckbox";

const TableBodyWithCheckbox = <T extends object>(props: TableBodyWithCheckboxProps<T>) => {
    const { onSelect, handleClickCheckbox, ...tableBaseBodyProps } = props;
    const {
        columns,
        data,
        border,
        hover,
        rowKey,
        withIndex,
        pagination,
        isSelected,
        hasDraftProperty,
    } = tableBaseBodyProps;

    return (
        <TableBaseBody
            customBody={
                <>
                    {data.map((record, dataIndex) => {
                        const key = getRowKey(record, rowKey) || dataIndex;

                        return (
                            <TableRowWithCheckbox
                                key={key}
                                columns={columns}
                                selected={isSelected?.(key)}
                                dataIndex={dataIndex}
                                record={record}
                                rowKey={rowKey}
                                border={border}
                                hover={hover}
                                onSelect={onSelect}
                                withIndex={withIndex}
                                pagination={pagination}
                                handleClickCheckbox={handleClickCheckbox}
                                hasDraftProperty={hasDraftProperty}
                            />
                        );
                    })}
                </>
            }
            {...tableBaseBodyProps}
        />
    );
};

export default TableBodyWithCheckbox;
