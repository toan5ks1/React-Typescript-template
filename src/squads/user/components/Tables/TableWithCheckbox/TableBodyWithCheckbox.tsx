import TableBaseBody from "src/components/Table/TableBaseBody";
import TableRowWithCheckbox from "src/components/Table/TableWithCheckbox/TableRowWithCheckbox";
import { TableBodyWithCheckboxProps } from "src/components/Table/table-types";
import { getRowKey } from "src/components/Table/utils";

const TableBodyWithCheckbox = <T extends object>(props: TableBodyWithCheckboxProps<T>) => {
    const { onSelect, handleClickCheckbox, ...tableBaseBodyProps } = props;
    const {
        columns,
        data,
        border,
        hover,
        rowKey,
        withIndex,
        pagination,
        isSelected,
        hasDraftProperty,
    } = tableBaseBodyProps;

    return (
        <TableBaseBody
            customBody={
                <>
                    {data.map((record, dataIndex) => {
                        const key = getRowKey(record, rowKey) || dataIndex;

                        return (
                            <TableRowWithCheckbox
                                key={key}
                                columns={columns}
                                selected={isSelected?.(key)}
                                dataIndex={dataIndex}
                                record={record}
                                rowKey={rowKey}
                                border={border}
                                hover={hover}
                                onSelect={onSelect}
                                withIndex={withIndex}
                                pagination={pagination}
                                handleClickCheckbox={handleClickCheckbox}
                                hasDraftProperty={hasDraftProperty}
                            />
                        );
                    })}
                </>
            }
            {...tableBaseBodyProps}
        />
    );
};

export default TableBodyWithCheckbox;
