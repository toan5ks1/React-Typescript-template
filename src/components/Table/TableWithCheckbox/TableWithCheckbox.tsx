import { Key, useCallback, useEffect, useMemo, useState } from "react";

import { differenceBy, intersectionBy, isEqual, uniqBy, xorBy } from "lodash";
import { arrayHasItem, safeStringify } from "src/common/utils/other";

import { Table } from "@mui/material";
import { useTheme } from "@mui/material/styles";

import TableBaseFooter from "../TableBaseFooter";
import TableWrapper from "../TableWrapper";
import { TableWithCheckboxProps } from "../table-types";
import { getBorder, getRowKey } from "../utils";
import TableBodyWithCheckbox from "./TableBodyWithCheckbox";
import TableHeaderWithCheckbox from "./TableHeaderWithCheckbox";

const TableWithCheckbox = <T extends object>(props: TableWithCheckboxProps<T>) => {
    const theme = useTheme();

    const {
        showHeader,
        styles,
        border = "all",
        columns,
        footer,
        body,
        tableProps,
        onSelect,
        component,
        data = [],
        listSelectedItems = [],
        errorMessage,
        withIndex,
        hasDraftProperty,
    } = props;

    const [listSelected, setSelected] = useState<T[]>(listSelectedItems);
    const dataWithDraft = hasDraftProperty
        ? data.filter((record) => record?.hasOwnProperty("isDraft"))
        : data;
    // Handle remove item in table => Re-check selected item
    useEffect(() => {
        setSelected(listSelectedItems);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [safeStringify(listSelectedItems)]);

    const handleSelectAllClick = (isSelectAll: boolean) => {
        if (isSelectAll) {
            const uniqMergeData = uniqBy(listSelected.concat(dataWithDraft), body.rowKey);

            setSelected(uniqMergeData);
            onSelect?.(uniqMergeData);

            return;
        }

        const leftData = differenceBy(listSelected, dataWithDraft, body.rowKey);
        setSelected(leftData);
        onSelect?.(leftData);
    };

    const handleClickCheckbox = (selectingRecord: T) => {
        const newSelectedRecords = listSelected.filter(
            (selectedRecord) => !isEqual(selectedRecord, selectingRecord)
        );

        // The filtered list has the same number of items so a new item is checked.
        if (newSelectedRecords.length === listSelected.length) {
            newSelectedRecords.push(selectingRecord);
        }

        setSelected(newSelectedRecords);
        onSelect?.(newSelectedRecords);
    };

    const handleCheckIsSelected = useCallback(
        (key: Key) => listSelected.some((record) => getRowKey(record, body.rowKey) === key),
        [body.rowKey, listSelected]
    );
    const numberOfSelectedItems = useMemo(
        () => intersectionBy(listSelected, dataWithDraft, body.rowKey).length,
        [dataWithDraft, listSelected, body.rowKey]
    );

    // Reset selected when data empty
    useEffect(() => {
        if (!body.loading && !dataWithDraft.length && listSelected.length) {
            setSelected([]);
        }
    }, [body.loading, dataWithDraft, listSelected]);

    // Handle when upsert data with same rowKey
    useEffect(() => {
        if (dataWithDraft && !body.loading && arrayHasItem(listSelected)) {
            const sameIDsData = intersectionBy(data, listSelected, body.rowKey);
            const extantData = xorBy(sameIDsData, listSelected, body.rowKey);

            setSelected([...extantData, ...sameIDsData]);
        }

        if (!arrayHasItem(listSelectedItems)) setSelected([]);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [safeStringify(dataWithDraft), safeStringify(listSelected)]);

    return (
        <TableWrapper component={component} errorMessage={errorMessage} styles={styles}>
            <Table style={styles?.table} {...tableProps}>
                <TableHeaderWithCheckbox
                    columns={columns}
                    border={getBorder(theme, border)}
                    visible={showHeader}
                    style={styles?.header}
                    select={
                        onSelect && {
                            onSelectAllClick: handleSelectAllClick,
                            hasData: !!dataWithDraft.length,
                            isCheckedAll: numberOfSelectedItems === dataWithDraft.length,
                            hasChecked:
                                numberOfSelectedItems > 0 &&
                                numberOfSelectedItems <= dataWithDraft.length,
                        }
                    }
                    withIndex={withIndex}
                />

                <TableBodyWithCheckbox
                    data={data} // to render all data include Draft
                    columns={columns}
                    border={getBorder(theme, border)}
                    onSelect={onSelect}
                    handleClickCheckbox={handleClickCheckbox}
                    isSelected={handleCheckIsSelected}
                    withIndex={withIndex}
                    hasDraftProperty={hasDraftProperty}
                    {...body}
                />
            </Table>
            {footer?.pagination && <TableBaseFooter {...footer} />}
        </TableWrapper>
    );
};

TableWithCheckbox.defaultProps = {
    columns: [],
    showHeader: true,
    border: "all",
    component: "div",
};

export default TableWithCheckbox;
