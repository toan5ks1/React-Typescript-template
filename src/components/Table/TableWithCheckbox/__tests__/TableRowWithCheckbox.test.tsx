import { PropsWithChildren } from "react";

import { TableColumn, TableRowWithCheckboxProps } from "../../table-types";
import TableRowWithCheckbox from "../TableRowWithCheckbox";
import { ItemTableBase } from "./TableWithCheckbox.test";

import { fireEvent, render } from "@testing-library/react";

const record: ItemTableBase = {
    id: 1,
    name: "Nam",
    email: "thanhnam@manabie.com",
};

const Wrapper = ({ children }: PropsWithChildren<{}>) => {
    return (
        <table>
            <tbody>{children}</tbody>
        </table>
    );
};

const renderWithWrapper = (props: TableRowWithCheckboxProps<any>) => {
    return render(<TableRowWithCheckbox {...props} />, {
        wrapper: Wrapper,
    });
};

describe("<TableRowWithCheckbox />", () => {
    const columns: TableColumn[] = [
        {
            key: "colName",
            title: "title 1",
            cellProps: {
                style: {
                    width: "60%",
                },
            },
        },
        {
            key: "colEmail",
            title: "title 2",
            cellProps: {
                style: {
                    width: "40%",
                },
            },
        },
    ];

    let props: TableRowWithCheckboxProps<ItemTableBase> = {
        columns,
        record,
        selected: true,
        dataIndex: 1,
        handleClickCheckbox: jest.fn(),
        border: {
            borderRight: `1px`,
            borderBottom: `1px`,
        },
        rowKey: "id",
    };

    it("should match snapshot", () => {
        const { container } = renderWithWrapper(props);
        expect(container).toMatchSnapshot();
    });

    it("should render with checkbox", () => {
        const { getByRole } = renderWithWrapper({ ...props, onSelect: () => {} });
        expect(getByRole("checkbox")).toBeInTheDocument();
    });

    it("should clickable on row", () => {
        const wrapper = renderWithWrapper({ ...props, onSelect: () => {} });
        fireEvent.click(wrapper.getByRole("checkbox"));
        expect(props.handleClickCheckbox).toBeCalled();
    });
});

describe("<TableRowWithCheckbox /> with render column", () => {
    const columns: TableColumn[] = [
        {
            key: "colName",
            title: "title 1",
            cellProps: {
                style: {
                    width: "60%",
                },
            },
            render: jest.fn(),
        },
        {
            key: "colEmail",
            title: "title 2",
            cellProps: {
                "data-testid": "Testing__iWantToTestAnotherAttr",
                style: {
                    width: "40%",
                },
            },
            render: jest.fn(),
        },
    ];
    let props: TableRowWithCheckboxProps<ItemTableBase> = {
        columns,
        record,
        selected: true,
        dataIndex: 1,
        handleClickCheckbox: jest.fn(),
        border: {
            borderRight: `1px`,
            borderBottom: `1px`,
        },
        rowKey: "id",
    };

    it("should render correct cell props", () => {
        const { container, getAllByTestId } = renderWithWrapper(props);
        expect(props.columns[0].render).toBeCalled();

        expect(container).toMatchSnapshot();

        expect(getAllByTestId("Testing__iWantToTestAnotherAttr")).toHaveLength(1);
    });
});
