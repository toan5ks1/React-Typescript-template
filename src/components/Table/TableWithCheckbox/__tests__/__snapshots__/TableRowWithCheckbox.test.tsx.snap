// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRowWithCheckbox /> should match snapshot 1`] = `
.emotion-0 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-0.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-0.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-0.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

.emotion-1 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  position: relative;
  border-right: 1px;
  border-bottom: 1px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  position: relative;
  border-right: none;
  border-bottom: 1px;
}

<div>
  <table>
    <tbody>
      <tr
        class="MuiTableRow-root TableBaseRow-selected Mui-selected emotion-0"
        data-testid="TableBase__row"
        data-value="1"
      >
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
          style="width: 60%;"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-2"
          />
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium emotion-3"
          style="width: 40%;"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-2"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<TableRowWithCheckbox /> with render column should render correct cell props 1`] = `
.emotion-0 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-0.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-0.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-0.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

.emotion-1 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  position: relative;
  border-right: 1px;
  border-bottom: 1px;
}

.emotion-2 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  position: relative;
  border-right: none;
  border-bottom: 1px;
}

<div>
  <table>
    <tbody>
      <tr
        class="MuiTableRow-root TableBaseRow-selected Mui-selected emotion-0"
        data-testid="TableBase__row"
        data-value="1"
      >
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
          style="width: 60%;"
        />
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium emotion-2"
          data-testid="Testing__iWantToTestAnotherAttr"
          style="width: 40%;"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
